common/include/pcl/impl/point_types.hpp

```diff
@@ -245,6 +247,19 @@ namespace pcl
     }; \
   };
 
+#define PCL_ADD_UNION_POINT3D \
```

Comment:
Since the union is only used once (in PointXY), I would say it is not necessary to create such a macro. Same for the other macros you added